src/java.desktop/share/classes/sun/java2d/loops/DrawGlyphListColor.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2021 JetBrains s.r.o.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package sun.java2d.loops;
+
+import sun.font.GlyphList;
+import sun.java2d.SunGraphics2D;
+import sun.java2d.SurfaceData;
+import sun.java2d.pipe.Region;
+
+import java.awt.*;
+
+/**
+ *   Draws color glyphs onto destination surface
+ */
+public class DrawGlyphListColor extends GraphicsPrimitive {
+
+    public final static String methodSignature =
+            "DrawGlyphListColor(...)".toString();
+
+    public final static int primTypeID = makePrimTypeID();
+
+    public static DrawGlyphListColor locate(SurfaceType srctype,
+                                            CompositeType comptype,
+                                            SurfaceType dsttype)
+    {
+        return (DrawGlyphListColor)
+            GraphicsPrimitiveMgr.locate(primTypeID,
+                                        srctype, comptype, dsttype);
+    }
+
+    protected DrawGlyphListColor(SurfaceType srctype,
+                                 CompositeType comptype,
+                                 SurfaceType dsttype)
+    {
+        super(methodSignature, primTypeID, srctype, comptype, dsttype);
+    }
+
+
+    public void DrawGlyphListColor(SunGraphics2D sg2d, SurfaceData dest,
+                                   GlyphList srcData,
+                                   int fromGlyph, int toGlyph) {
+        // actual implementation is in the 'General' subclass
+    }
+
+    // This instance is used only for lookup.
+    static {
+        GraphicsPrimitiveMgr.registerGeneral(
+                                new DrawGlyphListColor(null, null, null));
```

Comment:
If it is a complete "General" loop which supports all possible combination of incoming src/comp/dst/transform/etc I suggest to confirm that in the test.